### PR TITLE
Add windows build

### PR DESCRIPTION
**Pull request by hadim**

### CODE REVIEW

**Comment by jschueller** [comment]
you have to rerender for the appveyor config file to be generated

**Comment by hadim** [comment]
Thank you @jschueller

**Comment by hadim** [comment]
I don't understand the error for the Linux build... It was working before right ?

**Comment by jschueller** [comment]
You can ignore it for now, probably a faulty pinning.

**Comment by hadim** [comment]
Ok. `lapack` builds fine on my Windows test session so I hope it should work with AppVeyor.

(AppVeyor builds are so slow to start.... :-( )

**Comment by hadim** [comment]
Also on my Windows test I had to manually add the path to `ming32-make` (`C:\Program Files\Anaconda3\Library\mingw-w64\bin`). Any idea how I can do that in the `bld.bat` ?

**Comment by jschueller** [comment]
use ```%LIBRARY_PREFIX%\mingw-w64\bin```

File: recipe/bld.bat [hunk 1]
```diff
@@ -2,7 +2,7 @@ mkdir build
 cd build
 
 cmake -G "MinGW Makefiles" -DCMAKE_INSTALL_PREFIX=%PREFIX% -DCMAKE_BUILD_TYPE=Release -DLAPACKE=ON -DCBLAS=ON -Wno-dev ..
```
Comment (jschueller) [inline recipe/bld.bat]: should be ```CMAKE_INSTALL_PREFIX=%LIBRARY_PREFIX%```

**Comment by hadim** [comment]
Finally lapack build out of the box locally on Windows 10 and pass with circleci and travis. Let's wait for AppVeyor but I am confident. Hopefully it should be done before next year :-)

**Comment by hadim** [comment]
Looks like AppVeyor want to make me crazy...

**Comment by hadim** [comment]
I need some help here...

The windows x86 build fails when patching while patching works with x64...

The windows x64 fails because of the sh.exe/CMake error while I have properly added `set PATH=%PATH:C:/Program Files/Git/usr/bin/sh.exe;=%` before launching CMake. And this trick works locally on my Windows 10 test session.

Any help would be nice/awesome/appreciated . . .

**Comment by ocefpaf** [comment]
cc-ing our Windows heroes: @gillins, @patricksnape, and @msarahan here to take a look at https://github.com/conda-forge/lapack-feedstock/pull/3#issuecomment-257165040

File: recipe/meta.yaml [hunk 2]
```diff
@@ -9,22 +9,30 @@ source:
...
   number: 0
-  skip: true  # [win]
 
 requirements:
   build:
     - cmake
     - gcc  # [unix]
     - cloog 0.18.0 10  # [unix]
+    - m2w64-toolchain  # [win]
+    - m2-patch  # [win]
```
Comment (msarahan) [inline recipe/meta.yaml]: patch shouldn't be necessary.  Conda-build takes care of this one as an internal dependency.  were there errors without it?

File: recipe/meta.yaml [hunk 3]
```diff
@@ -9,22 +9,30 @@ source:
...
   run:
     - libgcc  # [unix]
+    - libgfortran  # [not win]
+    - m2w64-toolchain  # [win]
```
Comment (msarahan) [inline recipe/meta.yaml]: Not sure what runtime reqs are.  @mingwandroid - any ideas?

**Comment by mingwandroid** [comment]
`m2w64-gcc-libs` is what I'd use for Windows. All GCC, binutils and the GCC libs is a bit too heavy.

Personally I use `m2-patch`. The old `gnuwin32` executable is pretty buggy. I don't think that conda-build prescribes any particular patch package? I wanted to use a Python implementation but it wasn't up to scratch.

@hadim you should not need to add `C:\Program Files\Anaconda3\Library\mingw-w64\bin` to `PATH`. If like to get to the bottom of why you are needing to do that.

**Comment by hadim** [comment]
I have removed `m2-patch` and replaced `m2w64-toolchain` for `m2w64-gcc-libs` as runtime dependency.

@mingwandroid I don't understand what you're sating about `C:\Program Files\Anaconda3\Library\mingw-w64\bin`. If you talk about  `set PATH=%PATH:C:/Program Files/Git/usr/bin/sh.exe;=%` this to prevent a CMake bug that cannot run when `sh.exe` is in the path. It prevents the bug locally for me but not on AppVeyor.

**Comment by hadim** [comment]
For AppVeyor I am running in this weird random error for x86 : 

```
run_conda_forge_build_setup
no mem for new parser
Traceback (most recent call last):
  File "C:\conda\Scripts\conda-script.py", line 5, in <module>
    sys.exit(conda.cli.main())
  File "C:\conda\lib\site-packages\conda\cli\main.py", line 94, in main
    imported = importlib.import_module(module)
  File "C:\conda\lib\importlib\__init__.py", line 37, in import_module
    __import__(name)
MemoryError
```

See https://ci.appveyor.com/project/conda-forge/lapack-feedstock/build/1.0.31/job/occxp6qi9ckyjone.

And x64 seems to hang on `run_conda_forge_build_setup` : https://ci.appveyor.com/project/conda-forge/lapack-feedstock/build/1.0.31/job/m451o1fvpnrd13t1

**Comment by hadim** [comment]
The `MemoryError` does not stop the build actually. Now I am running with the same error for `x86` and `x64` related to CMake and git : 

```
CMake Error at C:/conda/envs/_build/Library/share/cmake-3.6/Modules/CMakeMinGWFindMake.cmake:22 (message):
  sh.exe was found in your PATH, here:
  C:/Program Files/Git/usr/bin/sh.exe
  For MinGW make to work correctly sh.exe must NOT be in your path.
  Run cmake from a shell that does not have sh.exe in your PATH.
  If you want to use a UNIX shell, then use MSYS Makefiles.
Call Stack (most recent call first):
  CMakeLists.txt:2 (project)
CMake Error: CMAKE_Fortran_COMPILER not set, after EnableLanguage
-- Configuring incomplete, errors occurred!
```

**Comment by hadim** [comment]
Look like it worked. Can someone review this and merge ? @jakirkham

**Comment by jschueller** [comment]
Would you mind keeping the rerendering part in a separate commit ? Otherwise it's ok for me and I'll merge this if no object in the following days.

**Comment by hadim** [comment]
Done. Let's wait Wednesday for any more comments before merging.

**Comment by hadim** [comment]
Could we merge this @jschueller?

**Comment by jschueller** [comment]
thanks

**Pull request merged**